File: main/src/main/java/com/crossapplication/main/repository/interfaces/ActivityRepository.java
package com.crossapplication.main.repository.interfaces;

import java.time.LocalDate;
import java.util.List;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import com.crossapplication.main.entity.Activity;

@Repository
public interface ActivityRepository extends JpaRepository<Activity, Long> {
    List<Activity> findByUserIdAndLogDateBetween(Long userId, LocalDate start, LocalDate end);

    @Query("SELECT COALESCE(SUM(a.caloriesBurned),0) FROM Activity a WHERE a.user.id = :userId AND a.logDate = :logDate")
    double sumCaloriesByUserIdAndLogDate(@Param("userId") Long userId, @Param("logDate") LocalDate logDate);

    List<Activity> findByUserIdAndLogDate(Long userId, LocalDate logDate);
}

File: flutter/flutterfrontend/lib/domain/repositories/activity_repository.dart
abstract class ActivityRepository {
  Future<List<Map<String, dynamic>>> findByUserIdAndLogDateBetween(
    int userId,
    DateTime start,
    DateTime end,
  );
  Future<double> sumCaloriesByUserIdAndLogDate(
    int userId,
    DateTime logDate,
  );
  Future<List<Map<String, dynamic>>> findByUserIdAndLogDate(
    int userId,
    DateTime logDate,
  );
  Future<Map<String, dynamic>?> findById(int id);
  Future<void> save(Map<String, dynamic> activity);
  Future<void> update(int id, Map<String, dynamic> activity);
  Future<void> delete(int id);
}
